lib/project-manager/src/main/scala/org/enso/projectmanager/control/core/CovariantFlatMap.scala

```diff
@@ -27,12 +29,29 @@ trait CovariantFlatMap[F[+_, +_]] {
     */
   def flatMap[E1, E2 >: E1, A, B](fa: F[E1, A])(f: A => F[E2, B]): F[E2, B]
 
+  /**
+    * `if` lifted into monad.
+    */
+  def ifM[E, B](
+    fa: F[E, Boolean]
+  )(ifTrue: => F[E, B], ifFalse: => F[E, B]): F[E, B] =
+    flatMap(fa)(if (_) ifTrue else ifFalse)
+
+  /**
+    * Keeps calling `f` until a `scala.util.Right[B]` is returned.
+    */
+  def tailRecM[E, A, B](a: A)(f: A => F[E, Either[A, B]]): F[E, B] =
```

Comment:
I'd call this `unfoldM`

Reply:
This name is used by cats library. So, I assumed that it is a standard in Scala.